src/controllers/hooks/router-store/index.tsx

```diff
@@ -18,3 +26,32 @@ export const useRouterActions = (): RouterActionsType => {
 
   return allActions;
 };
+
+export const useLocation = (): Location => {
+  const [routerState] = useRouterStore();
+
+  return routerState.location;
+};
+
+export const useHistory = (): BrowserHistory => {
+  const [routerState] = useRouterStore();
+
+  return routerState.history;
+};
+
+export const useParams = (): MatchParams => {
```

Comment:
If we expose something like this, let's make it more powerful: allow a white list of params to look after, so consumers re-render only when one of those params change. Also, should we expose some actions with it to update them and change the url?

Moreover, I would do the same thing for `useQueryParams()` so we can get that for query parameters. And would be cool to have actions to add/remove/update the query string too